api/client/client.go

```diff
@@ -838,6 +838,18 @@ func (c *Client) CreateUser(ctx context.Context, user types.User) error {
 	return trace.Wrap(err)
 }
 
+// CreateUserWithContext creates a new user from the specified descriptor.
+// TODO(tross) remove this once oss and e are converted to using the new signature.
+func (c *Client) CreateUserWithContext(ctx context.Context, user types.User) (types.User, error) {
+	err := c.CreateUser(ctx, user)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	created, err := c.GetUserWithContext(ctx, user.GetName(), false)
+	return created, trace.Wrap(err)
+}
```

Comment:
What happens if someone deletes the user between a successful create and the get?

What happens if the user has create permissions but not read permissions?